py-client/client_api.py

```diff
@@ -46,343 +61,390 @@ class AIAAClient:
+        :return: returns json containing list of models and details
         """
+        logger = logging.getLogger(__name__)
+        logger.debug('Fetching Model Details')
 
-        endpoint = "http://" + self.server_ip + ":" + self.server_port + "/" + self.api_version + "/" + api
-        url = endpoint
-        if args_dict:
-            sep = '?'
-            for key, value in args_dict.items():
-                url = url + sep + key + "=" + value
-                sep = '&'
+        selector = '/' + self.api_version + '/models'
+        if label is not None and len(label) > 0:
+            selector += '?label=' + AIAAUtils.urllib_quote_plus(label)
 
-        print("Connecting to: ", url)
+        response = AIAAUtils.get_method(self.server_url, selector)
+        response = response.decode('utf-8') if isinstance(response, bytes) else response
+        return json.loads(response)
 
-        if not params:
-            response = requests.get(url)
-        else:
-            data = {'params': params}
-            print(data)
-            if not files:
-                response = requests.post(url, data=data)
-            else:
-                response = requests.post(url, data=data, files=files)
-
-        return response
-
-    def model_list(self, result_file_prefix):
+    def segmentation(self, model, image_in, image_out):
         """
-        Get the current supported model list
-        :param result_file_prefix: File name to store the result
-        :return: returns json containing list of models and details supported for DEXTR3D
+        3D image segmentation using segmentation method
+        :param model: model name according to the output of model_list()
+        :param image_in: input 3D image file name
+        :param image_out: output files will be stored
+        :return: returns json containing extreme points for the segmentation mask
+
+        Output 3D binary mask will be saved to the specified file
         """
+        logger = logging.getLogger(__name__)
+        logger.debug('Preparing for Segmentation Action')
 
-        response = self._call_server("models")
-        file_names = _process_response(response, result_file_prefix)
-        print(file_names)
-        return file_names
+        selector = '/' + self.api_version + '/segmentation?model=' + AIAAUtils.urllib_quote_plus(model)
+        fields = {'params': '{}'}
+        files = {'datapoint': image_in}
 
-    # noinspection PyUnresolvedReferences
-    def dextr3d(self, model_name, image_file_path, result_file_prefix, point_set, pad=20,
-                roi_size='128x128x128', sigma=3):
+        logger.debug('Using Selector: {}'.format(selector))
+        logger.debug('Using Fields: {}'.format(fields))
+        logger.debug('Using Files: {}'.format(files))
+
+        form, files = AIAAUtils.post_multipart(self.server_url, selector, fields, files)
+        AIAAUtils.save_result(files, image_out)
+        return form
```

Comment:
form is a dict, not a json

So we should either do json.dumps or modify docstring

Reply:
json is actually a dict object.. so we can use vice-versa.. dumps will give string representation of a dict/json